Source/Plugin.BLE/Windows/Extensions/GattProtocolErrorParser.cs

```diff
@@ -0,0 +1,125 @@
+// See the LICENSE file in the project root for more information.
+
+
+/// <summary>
+/// Helper function when working with <see cref="GattProtocolError" />
+/// </summary>
+internal static class GattProtocolErrorParser
+{
+    /// <summary>
+    /// Helper to convert an Gatt error value into a string
+    /// </summary>
+    /// <param name="errorValue"> the byte error value.</param>
+    /// <returns>String representation of the error</returns>
+    public static string GetErrorString(this byte? errorValue)
+    {
+        var errorString = "Protocol Error";
+
+        if (errorValue.HasValue == false)
+        {
+            return errorString;
+        }
+
+        if (errorValue == GattProtocolError.AttributeNotFound)
+        {
+            return "Attribute Not Found";
+        }
+
+        if (errorValue == GattProtocolError.AttributeNotLong)
+        {
+            return "Attribute Not Long";
+        }
+
+        if (errorValue == GattProtocolError.InsufficientAuthentication)
+        {
+            return "Insufficient Authentication";
+        }
+
+        if (errorValue == GattProtocolError.InsufficientAuthorization)
+        {
+            return "Insufficient Authorization";
+        }
+
+        if (errorValue == GattProtocolError.InsufficientEncryption)
+        {
+            return "Insufficient Encryption";
+        }
+
+        if (errorValue == GattProtocolError.InsufficientEncryptionKeySize)
+        {
+            return "Insufficient Encryption Key Size";
+        }
+
+        if (errorValue == GattProtocolError.InsufficientResources)
+        {
+            return "Insufficient Resources";
+        }
+
+        if (errorValue == GattProtocolError.InvalidAttributeValueLength)
+        {
+            return "Invalid Attribute Value Length";
+        }
+
+        if (errorValue == GattProtocolError.InvalidHandle)
+        {
+            return "Invalid Handle";
+        }
+
+        if (errorValue == GattProtocolError.InvalidOffset)
+        {
+            return "Invalid Offset";
+        }
+
+        if (errorValue == GattProtocolError.InvalidPdu)
+        {
+            return "Invalid Pdu";
+        }
+
+        if (errorValue == GattProtocolError.PrepareQueueFull)
+        {
+            return "Prepare Queue Full";
+        }
+
+        if (errorValue == GattProtocolError.ReadNotPermitted)
+        {
+            return "Read Not Permitted";
+        }
+
+        if (errorValue == GattProtocolError.RequestNotSupported)
+        {
+            return "Request Not Supported";
+        }
+
+        if (errorValue == GattProtocolError.UnlikelyError)
+        {
+            return "UnlikelyError";
+        }
+
+        if (errorValue == GattProtocolError.UnsupportedGroupType)
+        {
+            return "Unsupported Group Type";
+        }
+
+        if (errorValue == GattProtocolError.WriteNotPermitted)
+        {
+            return "Write Not Permitted";
+        }
```

Comment:
IMHO this whole block here should better be one large `switch` statement than a long succession of `if` statements.

Reply:
I will rather not change this file, since it is  this file:
https://github.com/CommunityToolkit/WindowsCommunityToolkit/blob/main/Microsoft.Toolkit.Uwp.Connectivity/BluetoothLEHelper/GattProtocolErrorParser.cs
I have now added comments in the top of the file to clarify this.
The reason for not just including the `CommunityToolkit.WinUI.Connectivity` package is that this `GetErrorString` method is the only component/class, which is needed and the CommunityToolkit.WinUI.Connectivity is a monster IMO

Reply:
Ok then, fine with me.